pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -286,14 +285,13 @@ func extractServicePorts(gw gateway.Gateway) []corev1.ServicePort {
 		Port: int32(15021),
 	})
 	portNums := map[int32]struct{}{}
-	for i, l := range gw.Spec.Listeners {
+	for _, l := range gw.Spec.Listeners {
 		if _, f := portNums[int32(l.Port)]; f {
 			continue
 		}
 		portNums[int32(l.Port)] = struct{}{}
-		name := fmt.Sprintf("%s-%d", strings.ToLower(string(l.Protocol)), i)
 		svcPorts = append(svcPorts, corev1.ServicePort{
-			Name: name,
+			Name: string(l.Name),
```

Comment:
Is Name a required field ( validated, etc) ? Should we have a fallback if not set ?

Reply:
It is required, part of the CRD schema. Fallback is a good idea though, just in case..